<a href="https://colab.research.google.com/github/SauleRub/DI_Bootcamp/blob/main/Week15/Day4/DailyChallenge_Week15Day4.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Calculate the Required Sample Size:

In [1]:
# Re-import needed module after code execution environment was reset
from statsmodels.stats.power import NormalIndPower

# Initialize power analysis object
analysis = NormalIndPower()

# Parameters for sample size calculation
effect_size = 0.2   # Cohen's h for increase from 5% to 7%
alpha = 0.05        # Significance level
power = 0.8         # Desired power

# Calculate required sample size per group
sample_size = analysis.solve_power(effect_size=effect_size, power=power, alpha=alpha, ratio=1, alternative='two-sided')

sample_size

392.4430232577885

Analyze the Impact of Effect Size:

In [2]:
# List of effect sizes to analyze
effect_sizes = [0.1, 0.2, 0.3, 0.4]
sample_sizes = []

# Calculate required sample size for each effect size
for es in effect_sizes:
    ss = analysis.solve_power(effect_size=es, power=power, alpha=alpha, ratio=1, alternative='two-sided')
    sample_sizes.append(round(ss))

# Print the results
for es, ss in zip(effect_sizes, sample_sizes):
    print(f"Effect size {es} → Required sample size per group: {ss}")

Effect size 0.1 → Required sample size per group: 1570
Effect size 0.2 → Required sample size per group: 392
Effect size 0.3 → Required sample size per group: 174
Effect size 0.4 → Required sample size per group: 98


📚 Explanation (for the bakery team):

Imagine trying to detect a small difference in cookie taste (like 5.01 vs. 5.02 stars). You’d need to let lots of customers try both to even notice it.

That’s why:
	•	Smaller effects → more people needed to detect a difference.
	•	Larger effects → easier to detect, so fewer people are needed.

Think of it like spotting a sprinkle vs. a whole cherry on top — tiny changes require more careful tasting (aka larger sample), but big ones are obvious quickly.



Explain the Relationship:

🧁✨ Sweet Bytes: Power Up Your A/B Testing – Team Summary

Hey Team! 🎉

Before we roll out our brand new checkout process, we ran the numbers to make sure we’re making a smart, data-driven decision (because baking up sweet sales deserves precision!).

Here’s what we found:


---



🍪 How many tasters (aka users) do we need?

We expect our new checkout to boost conversion from 5% to 7% — that’s a small but sweet effect 🍰.
To detect this with 80% certainty (so we’re not just guessing), we need about 392 users per group — one group for the current checkout and one for the new one.



---

🧠 The smaller the change, the more users we need to tell whether it’s real or just a fluke.
Think of it like tasting a pinch of cinnamon vs. a burst of chocolate. Big changes stand out fast — small ones need more samples.


---

💡 Why this matters

Understanding this balance helps us:
	•	Avoid wasting time testing tiny changes with too few users
	•	Avoid missing big opportunities by stopping too soon
	•	Save money by testing smartly and confidently

Now we’re ready to bake decisions backed by real dough (data)!
Let’s launch our checkout test the right way. 🛒🍩📈